### PR TITLE
Use Tugboat for PR sandboxes

### LINKED ISSUE #4351: Use Tugboat.qa for PR sandboxes
**Description of the need**
We are considering using Tugboat.qa for PR sandboxes for Backdrop core (since that's Tugboat's advertised feature).

**Proposed solution**
Backdrop core will have a new, hidden directory with Tugboat-specific files inside. These will trigger preview sites to be built whenever a PR is opened. Previews will be automatically rebuilt when the PR is updated (e.g. with new commits). Comments will be posted to the PR when preview sites (sandboxes) are ready with login details so people can test the new functionality/bug fix.

**Alternatives that have been considered**
We are currently using Zen.ci for all of this, but there have been [some issues](https://github.com/backdrop-ops/backdropcms.org/issues/276) and so we'd like to see if Tugboat is a better fit...

**How to test this functionality:**

1. Fork BWPanda's Backdrop repo: https://github.com/BWPanda/backdrop (pretend this is the main Backdrop repo)
1. Clone and checkout the `tugboat` branch (pretend this is Backdrop's `1.x` branch)
1. Create a new branch (for the changes you want to make to Backdrop)
1. Make, commit and push your changes
1. Create a PR against `BWPanda:tugboat` (pretend you're making it against `backdrop:1.x`)
1. Wait for the Tugboat preview to be generated and displayed in the PR, then test it to see if your changes are visible
1. Report back here with your experience

This issue has been given the 'bug' milestone candidate label as it's hoped this can get into the next bugfix release. This exception is due to the fact that the proposed additional code won't affect Backdrop sites at all, only the repository/PRs (i.e. no chance of introducing bugs to existing Backdrop sites).

---

Advocate: @BWPanda 

**Status:** Code has been merged, now we just need to setup Tugboat, etc. (see https://github.com/backdrop/backdrop-issues/issues/4351#issuecomment-604087937).

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/4351

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Ignore this file so it can be changed 
+# Ignore this file so it can be changed
```
Comment (klonos) [inline .gitignore]:
```suggestion
# Ignore this file so it can be changed.
```

Comment (ghost) [reply .gitignore]: @klonos Thanks, committed.

**Comment by herbdool** [comment]
The other thing ZenCI does is run all the tests. I'm guessing we'll need that running in Tugboat before we can switch.

**Comment by ghost** [comment]
> The other thing ZenCI does is run all the tests. I'm guessing we'll need that running in Tugboat before we can switch.

I disagree. That's a completely seperate issue. We should be able to have existing tests running at the same time as Tugboat's preview site.

**Comment by herbdool** [comment]
Odd failing tests but we should make sure they're unrelated. Restarting

**Comment by ghost** [comment]
Merged into 1.x and 1.16.x.

**Pull request merged**